apps/nestjs-api/src/reminder-emails/reminder-emails.service.ts

```diff
@@ -201,6 +255,73 @@ export class ReminderEmailsService {
   //   return []
   // }
 
+  async getPendingMenteeApplications() {
+    const pendingMentorshipApplications =
+      await this.conMentorshipMatchesService.findAll({
+        Mentorship_Match_Age_In_Days__c: 14,
+        Status__c: MentorshipMatchStatus.APPLIED,
+      })
+
+    const reducedMatches = pendingMentorshipApplications.reduce(
```

Comment:
From this point onwards the function looks overly complicated. Can't we just do something like this?

```
allMentorsMentees = [
  ...pendingMentorshipApplications.map(x => x.props.menteeId),
  ...pendingMentorshipApplications.map(x => x.props.mentorId),
]
allMentorsMentees = deduplicate(allMentorsMentees)
allMentorsMentees = conProfilesService.findAll(blabla $in: allMentorsMentees)

pendingMentorshipApplications = pendingMentorshipApplications.map(
  application => {
    const mentor = allMentorsMentees.find(x => x.props.id === application.mentorId)
    const mentee = allMentorsMentees.find(x => x.props.id === application.menteeId)

    return {
      mentorEmail: mentor?.props.email,
      mentorFirstName: mentor?.props.firstName,
      menteeFirstName: mentee?.props.firstName,
      menteeFullName: mentee?.props.fullName,
      matchDate: formatDate(new Date(application['matchDate']), 'PPP'),
    }
  }
)
```

Reply:
Hey, did this actually get addressed? 🤔